tests/lax_scipy_sparse_test.py

```diff
@@ -98,13 +98,16 @@ def _fetch_preconditioner(self, preconditioner, A, rng=None):
     shape=[(4, 4), (7, 7)],
     dtype=[np.float64, np.complex128],
     preconditioner=[None, 'identity', 'exact', 'random'],
+    assume_ipart_is_zero=[False, True]
   )
-  def test_cg_against_scipy(self, shape, dtype, preconditioner):
+  def test_cg_against_scipy(self, shape, dtype, preconditioner, assume_ipart_is_zero):
     if not config.enable_x64.value:
       raise unittest.SkipTest("requires x64 mode")
 
     rng = jtu.rand_default(self.rng())
     A = rand_sym_pos_def(rng, shape, dtype)
+    if assume_ipart_is_zero:
+      A = A - jnp.imag(A)
```

Comment:
floating point math on accelerators can be weird, so I'd probably do `A = A.real.astype(A.dtype)` instead.